lldb/test/API/python_api/thread/TestThreadAPI.py

```diff
@@ -52,6 +52,7 @@ def test_negative_indexing(self):
         self.build()
         self.validate_negative_indexing()
  
+    @expectedFailureAll(oslist=["windows"], archs=["x86_64"])
```

Comment:
I am still seeing this test fail with:
```
======================================================================
FAIL: test_StepInstruction_dwarf (TestThreadAPI.ThreadAPITestCase.test_StepInstruction_dwarf)
    Test that StepInstruction preserves the plan stack.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1756, in test_method
    return attrvalue(self)
           ^^^^^^^^^^^^^^^
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 58, in test_StepInstruction
    self.step_instruction_in_called_function()
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 334, in step_instruction_in_called_function
    self.assertRegex(
AssertionError: Regex didn't match: '.*call_me\\(bool\\)$' not found in 'main' : Stopped in call_me(bool)
Config=x86_64-C:\workspace\llvm-project\build\bin\clang.exe
----------------------------------------------------------------------

Reply:
I recently updated this for Windows on Arm https://github.com/llvm/llvm-project/commit/3e06392c7db0eacfca94a176d430d9988b3ffbd6, that's where that regex came from.

We need to know if it's a cosmetic difference or whether it's stopped at a completely different place. Running the test with `lldb-dotest` and the `-t` flag will hopefully show you what it was doing.

Reply:
Thanks! I'll try that and update here.

Reply:
Here's the full output with 3e06392 cherry-picked. https://pastebin.com/LQ9QReBr

Reply:
If I change this line:

```
self.runCmd("expr -i 0 -- call_me(true)", check=False)
```
to this:

```
self.runCmd("expr -i 0 -- call_me(true)", check=True)
```

I see this error:

```
======================================================================
FAIL: test_StepInstruction_dwarf (TestThreadAPI.ThreadAPITestCase.test_StepInstruction_dwarf)
   Test that StepInstruction preserves the plan stack.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1761, in test_method
    return attrvalue(self)
           ^^^^^^^^^^^^^^^
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 16, in test_StepInstruction
    self.step_instruction_in_called_function()
  File "C:\workspace\llvm-project\lldb\test\API\python_api\thread\TestThreadAPI.py", line 291, in step_instruction_in_called_function
    self.runCmd("expr -i 0 -- call_me(true)", check=True)
  File "C:\workspace\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1006, in runCmd
    self.assertTrue(self.res.Succeeded(), msg if (msg) else CMD_MSG(cmd))
AssertionError: False is not true : Command 'expr -i 0 -- call_me(true)
Error output:
error: Couldn't look up symbols:
  void call_me(bool)
Hint: The expression tried to call a function that is not present in the target, perhaps because it was optimized out by the compiler.
' did not return successfully
Config=x86_64-C:\workspace\llvm-project\build\bin\clang.exe
----------------------------------------------------------------------

Reply:
Seems like my Clang invocation is generating different debug information than expected:

```
PS C:\workspace\llvm-project> llvm-pdbutil.exe pretty -all C:\workspace\llvm-project\build\lldb-test-build.noindex\python_api\thread\TestThreadAPI.test_StepInstruction_dwarf\a.pdb | Select-String -Pattern call_me

  func [0x00001000+ 0 - 0x00001025-37 | sizeof= 37] (FPO) void __cdecl call_me(bool should_spin)
  func [0x00001000+ 0 - 0x00001025-37 | sizeof= 37] (FPO) void __cdecl call_me(bool should_spin)
  public [0x00001000] ?call_me@@YAX_N@Z
```

Is the issue here that debug symbols have the additional `should_spin` parameter name, thus LLDB is not able to look them up since it is search for `void call_me(bool)`?

Reply:
I see the same function name/signature.
Btw, I'd recommend adding `--env;LLDB_USE_LLDB_SERVER=1` to `LLDB_TEST_USER_ARGS` cmake argument. Without it, lldb can't even stop on a breakpoint on my machine. It may be helpful if you haven't set it yet.